client/header/activity-panel/panels/inbox/index.js

```diff
@@ -66,32 +47,61 @@ class InboxPanel extends Component {
 		);
 	}
 
-	renderNotes() {
-		const { lastRead, notes } = this.props;
+	renderNotes( hasNotes ) {
+		const { isUndoRequesting, lastRead, notes } = this.props;
 
-		const validNotes = filter( notes, ( note ) => {
-			const { is_deleted: isDeleted } = note;
-			const noteActive = has( note, 'is_deleted' ) ? ! isDeleted : true;
-			return noteActive;
-		} );
+		if ( isUndoRequesting === 'undo-dismiss-all' ) {
+			return;
+		}
 
-		if ( validNotes.length === 0 ) {
+		if ( ! hasNotes ) {
 			return this.renderEmptyCard();
 		}
 
 		const notesArray = Object.keys( notes ).map( ( key ) => notes[ key ] );
 
-		return notesArray.map( ( note ) => (
-			<InboxNoteCard
-				key={ note.id }
-				note={ note }
-				lastRead={ lastRead }
-			/>
-		) );
+		return notesArray.map( ( note ) => {
+			const showPlaceholder = isUndoRequesting
+				? isUndoRequesting === note.id
+				: false;
```

Comment:
This variable looks to be unnecessary - can we just move `isUndoRequesting === note.id` into the `if()` below?

Reply:
Sounds good! I solved it in the commit [8b64bb9](https://github.com/woocommerce/woocommerce-admin/pull/4379/commits/8b64bb904ab47025dc75cda8040e01eb8c21156d)